Source/missiles.cpp

```diff
@@ -3944,12 +3944,10 @@ void MI_Lightctrl(int i)
 	int pn = dPiece[mx][my];
 	assert((DWORD)pn <= MAXTILES);
 
-	if (missile[i]._misource == -1) {
-		if ((Point { mx, my } != missile[i].position.start) && nMissileTable[pn]) {
+	if (missile[i]._misource != -1 || Point { mx, my } != missile[i].position.start) {

```

Comment:
Curious that `int p = missile[i]._misource;` is used in the damage calculation above, but `p` is never used again in the whole function.

Reply:
Fixed